clients/go/pkg/worker/jobPoller.go

```diff
@@ -92,6 +93,11 @@ func (poller *jobPoller) activateJobs() {
 	for {
 		response, err := stream.Recv()
 		if err != nil {
+			// no need to retry if the error was simply that we reached the end of the stream
+			if err == io.EOF {
```

Comment:
@Sijoma - wdyt? I wasn't sure why this wasn't there previously imo. I know you didn't write this, but you're the last one to touch it :upside_down_face: It seems fine for me to avoid the retry if the streamer was completed normally, and we'll just poll normally via the interval later down the line.